.circleci/getChangelogs.sh

```diff
@@ -32,23 +32,46 @@ function cloneRepository() {
     old_version=$2
     new_version=$3
     repository_url=$4
-    echo "[OK] $project_name($repository_url) : $old_version -> $new_version"
+    echo "[OK] checkout: $project_name($repository_url)"
     git clone "$repository_url" "$project_name" --quiet
 }
 
+function getOldDependencyVersion(){
+    old_version=$(getDependencyVersion $OLD_LOCK_FILE $1)
+    if [ "$old_version" == "" ]; then
+        # this means this dependency is new as there is no old_version as per the requirements file
+        # therefore we fetch the oldest tag from the git remote
```

Comment:
Does it make sense to fetch all commits since the very first tag? :thinking: I can be overkill to include what could be a very long commit history.

Perhaps we could simply include a line saying something like:

"Component X added as a new dependency"

It's just an idea. :)

Reply:
If that makes more sense from a changelog perspective, sure. What is your opinion?
I am happy to make that change :)

Reply:
I'm just worried that we'll be fetching **a lot** of commit lines each time we add a new dependency. I really think it's more useful to someone going through the changelog to understand that a new dependency was added. If we could also include the version of the dependency that was added, that would be even better.

Reply:
I agree with @pauloribeiro-codacy .. we should just fetch the oldest tag and do a print saying that a new component was added.
Next release will have more details.

Reply:
Agreed. I actually implemented this without checking with @pauloribeiro-codacy :) I agree that it could introduce a lot of stuff. As @paulopontesm said, the next release will have more details in the changelog.